services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,41 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  const void * itemToQueue = (const void *)event;
+  BaseType_t sendEventResult = xQueueSend(thermalMgrQueueHandle, itemToQueue, 0);
+  if (sendEventResult != pdTRUE) return ERR_CODE_QUEUE_FULL;
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t checkOverUnder;
+  checkOverUnder.type = THERMAL_MGR_EVENT_OVER_UNDER;
+
+  /* Add interrupt event to the queue */
+  thermalMgrSendEvent(&checkOverUnder);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  lm75bd_config_t config = *(lm75bd_config_t *)pvParameters;
+  float currTemp;
+  thermal_mgr_event_t nextEvent;
+
   while (1) {
-    
+    /* Receive next event and read temperature */
+    BaseType_t recResult = xQueueReceive(thermalMgrQueueHandle, &nextEvent, 0);
```

Comment:
this variable isnt need you can just check the xQueueReceive return value directly in an if statement

Reply:
Also the return value should be checked since right now you are continuing and doing the rest of the code even if `xQueueReceive` returns pdFALSE, and since your block time is 0, you are essentially performing the code in an infinite loop instead of waiting for an event

Reply:
Also the block time should be set higher since we know that we'll only be reading temperature every 5 seconds anyways and this task has nothing else to do if we don't receive anything, there's no reason to spend CPU time polling the xQueueReceive even though it's empty. When an item enters the queue this task will become unblocked anyways